types/basetypes/dynamic_value.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package basetypes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+var (
+	_ DynamicValuable = DynamicValue{}
+)
+
+// DynamicValuable extends attr.Value for dynamic value types. Implement this interface
```

Comment:
Probably not for here, but perhaps the docs can provide some example(s) of where it would be useful to have a custom DynamicValue type?